internal/safeconfig/safeconfig.go

```diff
@@ -0,0 +1,50 @@
+package safeconfig
+
+import (
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"gopkg.in/yaml.v2"
+)
+
+type SafeConfig struct {
+	Domains []string `yaml:"domains"`
+}
+
+func New(pathToFile string) (SafeConfig, error) {
+	cfg := SafeConfig{}
+	if pathToFile == "" {
+		log.Debug().Msg("config file path is empty, skip loading")
+		return cfg, nil
+	}
+
+	if err := cfg.Reload(pathToFile); err != nil {
+		return cfg, err
+	}
+	return cfg, nil
+}
+
+func (cfg *SafeConfig) Reload(pathToFile string) error {
+	log.Info().Msgf("trying to load config from file %s", pathToFile)
+
+	filename, err := filepath.Abs(pathToFile)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get absolute path of file %s", pathToFile)
+	}
+	log.Debug().Msgf("absolute path of config file is %s", filename)
+
+	yamlFile, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return errors.Wrap(err, "failed to read file")
```

Comment:
```suggestion
		return fmt.Errorf("failed to read file: %w", err)
```

Reply:
It's better to use errors.Wrap, because fmt.Errorf will hide all information about underlying error. Sometimes it becomes problematic for cases when you would like to get stacktrace for underlying error.
It's not important here, may be, but, I think, it's more convinient way to wrap errors. If you wish, I will accept your changes :)

Reply:
ah indeed, let's leave it with pkg/errors then